### ISSUE TITLE
Performance report returns results not available due to invalid wpcom_performance_report_url option value

### ISSUE THREAD

**Issue by mgozdis** [Automattic/wp-calypso#97206]
### Quick summary

The value for the `wpcom_performance_report_url` option is sometimes invalid, which causes the API to return a 403 response and a `Results not available message` as shown here: ![AwvmnO.png](https://github.com/user-attachments/assets/17b73737-364e-4d6f-af53-3f54e6c82105)

Rather than the option value containing the domain name with the hash, it saved the domain name as `false`, such as `http://false&hash=hashvalue`

Screenshot: 

![Image](https://github.com/user-attachments/assets/7d35e025-21fc-4a03-8346-546cb71f17c1)

User report: 9148283-zd-a8c

### Steps to reproduce

It is not currently known how to reproduce, however, a button to launch the site was also displaying sporadically despite the site being launched and public. Clicking the button did not change the behavior of the report not displaying. 

### What you expected to happen

The report should display. 

### What actually happened

Results not available is returned.

### Impact

One

### Available workarounds?

Yes, easy to implement

### If the above answer is "Yes...", outline the workaround.

Delete the `wpcom_performance_report_url` option from the `wp_options` table and flush cache. The report will then generate and the option should be recreated with a valid value. 

You can also delete the `wpcom_performance_report_url` option by using `wp option delete wpcom_performance_report_url` command in CLI.

**NOTE:** Please run `wp option get wpcom_performance_report_url` to check the value first before deleting and share in any additional reports.

### Platform (Simple and/or Atomic)

Atomic

**Comment by Robertght** [comment]
@mgozdis I added the Droplr screenshot to the original report as links expire.

As we don't have specific steps to replicate, I'm also curious what can break this.

@Automattic/quake is this something your team can investigate?

**Comment by vykes-mac** [comment]
Thanks @mgozdis for reporting this. does it happens even if the user re-run the tests?

**Comment by mgozdis** [comment]
@vykes-mac Yes, re-running produced the same error. The only way I was able to resolve was by deleting the `wpcom_performance_report_url` in the database. 

I'll note that they or another HE also attempted testing this on a staging site. The same issue existed there since it syncs the data exactly as is. 

That could potentially introduce other issues when syncing staging/production sites if it ends up using a value for the wrong site. We may want to preserve that value from being overwritten during syncing, but that's a separate issue that would need to be tested.

**Comment by vykes-mac** [comment]
Thank you @mgozdis , I am able to reproduce on staging sites. Maybe that's where the error first occurred and was further synced to the production site.

**Comment by mgozdis** [comment]
With the site that reported it, I don't think they ever synced staging back to production. They deleted the staging site already so there are no logs. It may be an issue with how the value is initially saved and potentially reproducible if a site is not launched, private, etc. 

For preserving values when syncing, this may be helpful to read as there is a `Protected_Options_Action` class: D166504-code

**Comment by vykes-mac** [comment]
> if a site is not launched, private, etc.

If a site is not launch or is private, performance test is not executed. 

![Image](https://github.com/user-attachments/assets/c884dbb5-b3c6-458e-86f2-cba9945460a9)

**Comment by mgozdis** [comment]
On the site that reported it, that launch your site info would "flash" for a split second before the error even though the site was launched and public. They had `Prevent third-party sharing` enabled, but changing that and making the site private or coming soon and back to public had no effect either.

**Comment by vykes-mac** [comment]
I'm not sure how to reproduce this, but from the error I can see that the report url was saved `http://false&hash=hashvalue` , the false suggest there was an error with the url return from the backend, in this case we shouldn't have saved that url but allow the user to re-run the test. I see in the [code](https://github.com/Automattic/wp-calypso/blob/8a0fabd387612d9ac1895f9518e5a00ae20eec0e/client/hosting/performance/site-performance.tsx#L67) where the url is saved even though "false" is returned as the url. This should be fixed to prevent this issue and so the user will be able to retry without a faulty url.

**Comment by aisajib** [comment]
This came up on 9198686-zd-a8c too, which @ajaykj fixed.

**Comment by upwardmomentum84** [comment]
Another occurrence of this here: p1736781440657919-slack-C03TY6J1A

**Comment by vykes-mac** [comment]
I'm going to add a guard to prevent these scenarios and merge that code

**Comment by vykes-mac** [comment]
I've added a `guard` to prevent this issue [here](https://github.com/Automattic/wp-calypso/pull/98295) . Please feel free to reopen if we continue to get reports

**Comment by kelasante** [comment]
> I've added a `guard` to prevent this issue [here](https://github.com/Automattic/wp-calypso/pull/98295) . Please feel free to reopen if we continue to get reports

Another report in 9372142-zen

**Comment by vykes-mac** [comment]
@kelasante Thanks for reporting. It would have been great if we could see the value that was in the `wpcom_performance_report_url` to help see why it was invalid.

**Comment by kelasante** [comment]
> [@kelasante](https://github.com/kelasante) Thanks for reporting. It would have been great if we could see the value that was in the `wpcom_performance_report_url` to help see why it was invalid.

Thanks for sharing this. I'll keep that in mind if I encounter other users with this issue.

**Comment by supernovia** [comment]
9414660-zd-a8c

**Comment by cat-og** [comment]
9420953-zd-a8c

**Comment by vykes-mac** [comment]
Leaving this note here for anyone who reports an error in the future. To help us fix the issue can you try and get a snapshot of the `wpcom_performance_report_url` for sites having the issue or any other information that can help identify the underlying cause. Thanks much.

**Comment by cat-og** [comment]
> Leaving this note here for anyone who reports an error in the future. To help us fix the issue can you try and get a snapshot of the `wpcom_performance_report_url` for sites having the issue or any other information that can help identify the underlying cause. Thanks much.

Sorry, just checking in. I'm not sure how to do that, is that the command in CLI that we should use to get this information before running the fix?

**Comment by vykes-mac** [comment]
@cat-og Yes, before deleting the option using the CLI you can `wp option get wpcom_performance_report_url` to see what's the value of the url.

**Comment by cat-og** [comment]
9391751-zd-a8c 
- This user was affected a few days ago and this issue has already been resolved. We did not have a chance to run the `option get wpcom_performance_report_url` first.

**Comment by vykes-mac** [comment]
Thank you @ajaykj for reporting the url, that provided good insight into how we can fix this issue

**Comment by paulopmt1** [comment]
Is this completed @vykes-mac?

**Comment by vykes-mac** [comment]
> Is this completed @vykes-mac?

Yeah that last PR should resolve this. We can reopen if there are new reports

**Comment by mdtanjid0** [comment]
I have a user who is facing the same issue: 9501208-zd-a8c on a staging site. 


CLI: `wp option get wpcom_performance_report_url`

Value:

`http://false&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6NzI0MDV9.XoqYRJjDLBMV_24-Uc62tJoYXelOfEjF7n2tVwasjw0`

Deleting that fixes the issue. 

@paulopmt1 Could you please check if we need to reopen this? 

Thanks!

**Comment by paulopmt1** [comment]
> I have a user who is facing the same issue: 9501208-zd-a8c on a staging site.

@vykes-mac can you look into this new case to define the next steps?

**Comment by vykes-mac** [comment]
> > I have a user who is facing the same issue: 9501208-zd-a8c on a staging site.
> 
> @vykes-mac can you look into this new case to define the next steps?

@paulopmt1 I wonder if it's a case where this bad url was already on the site, also isn't staging sites unlaunched sites ? Is the performance runs on staging sites ?

I would say not to reopen issue as yet, if we continue to get more reports then we can look further into it.

**Comment by paulopmt1** [comment]
> I wonder if it's a case where this bad url was already on the site

I see. In that case, could we run a search in the entire database looking for the `wpcom_performance_report_url` that contains the `http://false&` pattern and remove those as a fix for all possible future cases of this issue?

**Comment by ajaykj** [comment]
Got another report today in p1746015199212229-slack-C03TY6J1A. wp option get wpcom_performance_report_url showed

```
http://false&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6NjkxNzB9.3-dNlfvSEirSrshCuw5QIFvtzJlD8P5SGHi5M0kgfXo
```

However, without us doing anything (not sure if the user did something), the value changed to 

```
https://myteams.co/&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6MTA5ODExfQ.51RXCLRgpEGhaffIrinXau1rRcHjy9XaXhEYQLPpDwE
```

With this change, the issue solved. I am reopening this report in case it was an edge case.

**Comment by wojtekn** [comment]
It was reopened due to another issue reported which was causing similar visible issue. We fixed it.
